src/bin/ion/commands/generate/generator.rs

```diff
@@ -380,377 +575,235 @@ impl<'a, L: Language + 'static> CodeGenerator<'a, L> {
-                                if let AbstractDataType::Sequence {
-                                    element_type: nested_element_type,
-                                    sequence_type: nested_sequence_type,
-                                } = &nested_type.abstract_data_type
-                                {
-                                    code_gen_context.abstract_data_type =
-                                        Some(AbstractDataType::Sequence {
-                                            element_type: nested_element_type.to_owned(),
-                                            sequence_type: nested_sequence_type.to_owned(),
-                                        });
-                                }
+                        // If the code gen type is not defined then we need to match with the given ISL type name and
+                        // add a new code gen type based on that.
+                        let mut name = vec![];
+                        if let Some(namespace) = &self.current_type_fully_qualified_name {
+                            name = namespace.to_vec();
+                        }
 
-                                // remove this nested type from the list as it will now be part of this field without generating separate nested type.
-                                nested_types.pop();
-                                return Ok(());
+                        let abstract_data_type = match isl_type.name().as_str() {
+                            "list" => {
+                                if data_model_node.name.contains("NestedType") {
+                                    AbstractDataType::Sequence(Sequence {
+                                        name: vec![],
+                                        doc_comment: None,
+                                        element_type: None,
+                                        sequence_type: Some(SequenceType::List),
+                                        source: parent_isl_type.to_owned(),
+                                    })
+                                } else {
+                                    AbstractDataType::WrappedSequence(WrappedSequence {
+                                        name: FullyQualifiedTypeReference {
+                                            type_name: name,
+                                            parameters: vec![],
+                                        },
+                                        doc_comment: None,
+                                        element_type: None,
+                                        sequence_type: Some(SequenceType::List),
+                                        source: parent_isl_type.to_owned(),
+                                    })
+                                }
                             }
-                        }
+                            "sexp" => {
+                                if data_model_node.name.contains("NestedType") {
```

Comment:
I think `data_model_node` needs an `is_nested` flag so we don't have to depend on the contents of the `name`.